### PR TITLE
add and populate 'referenced_by' property to ermrest_model Table

### PR DESCRIPTION
**Pull request by robes**

### CODE REVIEW

File: deriva/core/ermrest_model.py [hunk 1]
```diff
@@ -19,6 +19,23 @@ class Model (_ec.CatalogConfig):
     """
     def __init__(self, model_doc, **kwargs):
         super(Model, self).__init__(model_doc, **_kwargs(**kwargs))
+        Model.update_referenced_by(self)
+
+    @classmethod
```
Comment (karlcz) [inline deriva/core/ermrest_model.py]: I don't understand why this is a classmethod.  It should be a regular method taking `self` which is the model being updated.

File: deriva/core/ermrest_model.py [hunk 2]
```diff
@@ -109,6 +126,7 @@ def __init__(self, sname, tname, table_doc, **kwargs):
         super(Table, self).__init__(sname, tname, table_doc, **_kwargs(**kwargs))
         self.comment = table_doc.get('comment')
         self.kind = table_doc.get('kind')
+        self.referenced_by = []
```
Comment (karlcz) [inline deriva/core/ermrest_model.py]: This should probably be using the MultiKeyedList class like the sibling `keys` and `foreign_keys` fields in the parent class from ermrest_config

**Comment by robes** [comment]
@karlcz this has been updated when you have a chance to look at it again

**Comment by robes** [comment]
Yeah, I've got a script in deriva-commons for ETL'ing all fo the vocabulary tables and this feature is used by it.

**Pull request merged**